src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -18,16 +18,68 @@
 
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/core/DataModelTypes.h>
-#include <lib/core/GroupId.h>
-#include <lib/support/EnforceFormat.h>
-#include <lib/support/logging/Constants.h>
-#include <stdarg.h>
+#include <lib/support/CHIPMemString.h>
+
 #include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 
 namespace chip {
 
+/**
+ * Represents a key used for addressing a specific storage element.
+ *
+ * May contain generic fixed keys (e.g. "g/fidx") or formatted fabric-specific
+ * keys ("f/%x/..." where %x is the fabric index).
+ */
+class StorageKey
```

Comment:
I kind of inherited from existing logic ... this says "DefaultStorageKeyAllocator" which seems to imply it allocates storage keys with default values (hence the name storage key).

Ideally we would namespace it or find different names ... I am unsure what default here even means as it implies there may be a non-default version here, but the class as we have it today is non-overridable. Going and updating the entire logic seems a bit much though. I am claiming for now that this is a net improvement but not trying to fix everything.

What are your thoughts on this:
   - will rename (soon, starting now after typing this). However would `DefaultStorageKey` or `DefaultStorageKeyName` be a better name?
   - Is this api even sane? Should `DefaultStorageKeyAllocator` be a namespace since it is a class with all-static content? if a namespace, should the `storage key` be under that namespace?

Reply:
Overall, changes to storage API without a strategic design, at this point, should be minimized. We don't want to paint ourselves in a corner without a sufficient analysis. Your change is small enough right now, that it doesn't change the semantics significantly